pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,404 @@ import (
-	}
+		Expect(len(fn)).To(Equal(len("testfile.img") + len(wantString)))
+	})
+})
 
-	return ds, testBytes, nil
-}
+var _ = Describe("Streaming Data Conversion", func() {
 
-func getFileSize(testfile string) (int, error) {
-	f, err := os.Open(getTestFilePath(testfile))
-	defer f.Close()
+	tinyCoreIso := "tinyCore.iso"
+	baseTestImage, err := filepath.Abs(filepath.Join(TestImagesDir, tinyCoreIso))
 	if err != nil {
-		return 0, fmt.Errorf("Unable to open source datastream file %s", getTestFilePath(testfile))
+		Fail(fmt.Sprintf("Error getting abs path: %v\n", err))
 	}
-	fi, err := f.Stat()
-	if err != nil {
-		return 0, fmt.Errorf("Unable to stat file %v", f.Name())
-	}
-
-	return int(fi.Size()), nil
-}
-
-func getTestFilePath(testfile string) string {
-	// CWD is set within go test to the directory of the test
-	// being executed, so using relative path
-	imageDir, _ := filepath.Abs(TestImagesDir)
-	return filepath.Join(imageDir, testfile)
-}
 
-func getURLPath(testfile string) string {
-	// CWD is set within go test to the directory of the test
-	// being executed, so using relative path
-	imageDir, _ := filepath.Abs(TestImagesDir)
-	return "file://" + imageDir + "/" + testfile
-}
+	var tmpTestDir string
+	BeforeEach(func() {
+		tmpTestDir = testDir(os.TempDir())
+		if err != nil {
+			Fail(fmt.Sprintf("[BeforeEach] Failed created test dir: %v\n", err))
+		}
+		By(fmt.Sprintf("[BeforeEach] Creating temporary dir %s", tmpTestDir))
+		syscall.Umask(0000)
+		err = os.Mkdir(tmpTestDir, 0777)
+		if err != nil {
+			Fail(fmt.Sprintf("Could not create tmp file: %v\n ", err))
+		}
+	})
 
-func startHTTPServer(port int, dir string) (*http.Server, error) {
-	server := &http.Server{
-		Addr:    fmt.Sprintf(":%d", port),
-		Handler: http.FileServer(http.Dir(dir)),
-	}
+	AfterEach(func() {
+		By(fmt.Sprintf("[AfterEach] Cleaning up temporary dir %s", tmpTestDir))
+		os.RemoveAll(tmpTestDir)
+	})
 
-	go func() {
-		if err := server.ListenAndServe(); err != nil {
-			// cannot panic, because this probably is an intentional close
-			log.Printf("Httpserver: ListenAndServe() error: %s", err)
+	table.DescribeTable("when data is in a supported file format", func(originalFile string, useVirtSize bool, expectFormats []string) {
+		By(fmt.Sprintf("Getting size of source file %q", originalFile))
+		finfo, err := os.Stat(originalFile)
+		Expect(err).NotTo(HaveOccurred())
+		sourceSize := finfo.Size()
+		fmt.Fprintf(GinkgoWriter, "INFO: size = %d\n", sourceSize)
+
+		By(fmt.Sprintf("Converting source file to format: %s", expectFormats))
+		// Generate the expected data format from the random bytes
+		testSample, err := utils.FormatTestData(originalFile, tmpTestDir, expectFormats...)
+		Expect(err).NotTo(HaveOccurred(), "Error formatting test data.")
+		fmt.Fprintf(GinkgoWriter, "INFO: converted source file name is %q, in dir %q\n", testSample, tmpTestDir)
+
+		tempTestServer := createTestServer(filepath.Dir(testSample))
```

Comment:
why is a new http file server needed for each iteration of this test?

Reply:
httptest servers are cheap to start (I am creating a new one for each test in previous tests as well)